Sources/NIOHTTP1Client/main.swift

```diff
@@ -0,0 +1,100 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+import NIOHTTP1
+import NIOOpenSSL
+import Foundation
+
+private final class HTTPResponseHandler: ChannelInboundHandler {
+
+    let promise: EventLoopPromise<Void>
+
+    init(_ promise: EventLoopPromise<Void>) {
+        self.promise = promise
+    }
+
+    typealias InboundIn = HTTPClientResponsePart
+
+    func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+        let httpResponsePart = unwrapInboundIn(data)
+        switch httpResponsePart {
+        case .head(let httpResponseHeader):
+            print("\(httpResponseHeader.version) \(httpResponseHeader.status.code) \(httpResponseHeader.status.reasonPhrase)")
+            for (name, value) in httpResponseHeader.headers {
+                print("\(name): \(value)")
+            }
+        case .body(var byteBuffer):
+            if let responseBody = byteBuffer.readString(length: byteBuffer.readableBytes) {
```

Comment:
this can fail with UTF-8 strings (let's assume the server sends `"🎉"` in chunks of one byte, that would be `f0`, `9f`, `8e`,  `89` then this would print five times `⍰` instead of 🎉.)

but I might be fine with that, CC @Lukasa

Reply:
Yeah, maybe we should skip `print` and instead write bytes to stdout?

Reply:
sounds like a plan to me.

Reply:
Thanks! I removed `print` and use stdout in 39b519a.